Глобальные улучшения

js-denton opened the pull request:
Было добавлено множество изменений:
* Исправлены опечатки
* Добавлены заглавные буквы
* Улучшена нумерация
* Сборка книги
* Сноски переделаны на стиль markdown
* Удалены пустые пробелы
* Добавлена сборка fb2

На настольной ОС проблемы с открытием epub/fb2. Корректно открываются в calibre (использую Fedora)

js-denton committed:
Removed trailing spaces

js-denton committed:
Added Capitalizer and UPPERCASE

* Many sentences started with a small letter
* Proper names and terms started with a samll letter
* Caption to the drawings started with a small letter

js-denton committed:
Fixed enumeration

js-denton committed:
Fixed typos

js-denton committed:
Fixed sentences

* Stilistic
* Semantic

js-denton committed:
Changed all footnote on markdown style

ligurio commented on style.css:
```diff
@@ -1,74 +0,0 @@
-body {
```
Why removed?

"* Removed css file because default style decide better results" - this is debatable

js-denton replied:
<details> 
  <summary>По поводу стилей </summary>
    (Сверху вниз)
    1. Оригинал
    2. Стиль по умолчанию
    3. Стиль который сейчас в upsteram

![Снимок экрана от 2024-01-17 22-08-16](https://github.com/ligurio/swebok-2004-in-russian/assets/12989870/f5b59688-63db-412e-a143-bfc493f08abc)
![Снимок экрана от 2024-01-17 22-07-25](https://github.com/ligurio/swebok-2004-in-russian/assets/12989870/ff86a172-2ecd-4aed-ae1e-37a16763707b)
![Снимок экрана от 2024-01-17 22-07-19](https://github.com/ligurio/swebok-2004-in-russian/assets/12989870/85f83b26-b959-4f11-a20a-9cbb89024ac9)
</details>

Шрифт в оригинале подобен FreeSerif. Pandoc по умолчанию тоже [использует serif](https://github.com/jgm/pandoc/blob/18f0b70d40e18c2c5b34c20a45f7b625a917b5a4/data/epub.css#L20C12-L20C12)
Как я понимаю это стандарт в печатном производстве.
С нумерованными списками я проблем не заметил.
У style.css как минимум есть проблемы с выравниванием текста и указателями на сноски.
Классно что после заголовка есть разделитель, но цвет заголовка как нажатая гиперссылка (по мне странное решение).

Редактировать style.css показалось не целесообразным так как стиль по умолчанию в pandoc даёт лучший результат.

Предлагаю использовать стиль pandoc по умолчанию или на его основе создать новый по указанным требованиям.

ligurio replied:
> Шрифт в оригинале подобен FreeSerif. Pandoc по умолчанию тоже [использует serif](https://github.com/jgm/pandoc/blob/18f0b70d40e18c2c5b34c20a45f7b625a917b5a4/data/epub.css#L20C12-L20C12)

Georgia и serif, если быть точным.

Спасибо за скрины, вцелом я согласен, дефолтный вариант не такой уж и плохой.

ligurio commented on Makefile:
```diff
@@ -13,18 +13,21 @@ MAKRDOWN_FILES += software_lifecycle_models.markdown
 MAKRDOWN_FILES += bibliography.markdown
 
 PANDOC = pandoc
-PANDOC_OPT += -s --toc-depth=2 --toc --self-contained
```
What for these changes?

js-denton replied:
--self-contained - deprecated https://pandoc.org/MANUAL.html#options-affecting-specific-writers
-s нужен только для html. При сборке epub этот параметр используется автоматически https://pandoc.org/MANUAL.html#general-writer-options
Глубину формирования оглавления оставил по умолчанию. Мне показалось что уровень вложенности 2 слишком мал (как я понял по умолчанию используется 3). В оригинале тоже вложенность 3. https://ieeecs-media.computer.org/media/education/swebok/swebok-v3.pdf

Соответственно по итогу оставил только генерацию оглавления.

ligurio replied:
Вынесите это изменение в отдельный коммит и в нем опишите причины изменения. Не делайт, пожалуйста, разные изменения в одном коммите.

js-denton replied:
Есть https://github.com/ligurio/swebok-2004-in-russian/pull/2/commits/f30b87e3607a8689b6d199b3976ac6cf3af99385

ligurio commented on .gitignore:
```diff
@@ -1,3 +1,4 @@
 swebok_2004_ru.epub
```
typo in short commit message: Improvment -> Improvement

js-denton replied:
Исправил: https://github.com/ligurio/swebok-2004-in-russian/pull/2/commits/54648194e1a00d9c187981643f48301dfc84db7a

ligurio commented on 1_software_requirements.markdown:
```diff
@@ -62,26 +62,26 @@ SWEBOK охватывает не только вопросы структури
 
 ### Определение требований (Definition of a Software Requirement)
 
-– в данном случае подразумевается не процесс определения (извлечения, сбора, формирования,
```
it is not clear why hyphens were removed

js-denton replied:
Дефис удалён по причине того что перед этим указан заголовок. Начинать абзац с дефиса не очень хорошо. Кстати в оригинале тоже никакого дефиса нет.
Можно заменить "В данном случае" на "В сущности" или "По большому счёту".

ligurio replied:
Принято.

ligurio commented on software_lifecycle_models.markdown:
```diff
@@ -517,18 +517,19 @@ ISO/IEC 15288 (выпущен в 2002 году), фокусирующийся н
-1. Большая часть этих рисков связана с организационными и процессными аспектами
+8. Недостатки в работах, выполняемых внешними (по отношению к проекту) ресурсами.
+9. Недостаточная производительность получаемой системы.
+10. “Разрыв” в квалификации специалистов разных областей знаний.
```
What for?

github's markdown allows enumerated lists with `1.` on every line, see [^1]

[^1]: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#lists

js-denton replied:
Да, я знаю что можно писать одни единички и всё будет хорошо.
markdown-mode в emacs автоматически делает такое исправление при сохранении файла (как и удаление пустых пробелов). Так что я руками тут не правил :-).

ligurio replied:
тогда откатывайте изменение

js-denton replied:
Изменения нумерации откатаны

ligurio commented on 10_software_quality.markdown:
```diff
@@ -522,13 +522,6 @@ V&V-отчетов, а также различных типов планов –
 в идентификации расхождений с утвержденными спецификациями и стандартами.” -
 IEEE 1028-97 “IEEE Standard for Software Reviews”.
 
-#### Технические оценки (Technical Reviews)
-
-“Назначением технических оценок является исследование программного продукта для
-определения его пригодности для использования в надлежащих целях. Цель состоит
-в идентификации расхождений с утвержденными спецификациями и стандартами.” -
-IEEE 1028-97 “IEEE Standard for Software Reviews”.
-
```
Why this removed?

js-denton replied:
Это дубликат. Выше по тексту то же самое.

ligurio replied:
В отдельный коммит и укажите причину в commit message

js-denton replied:
Есть https://github.com/ligurio/swebok-2004-in-russian/pull/2/commits/46fe08ec37c6e8a70dee4ddd2c9fb0dd5d8a0fb5

ligurio replied:
спасибо

ligurio commented on 10_software_quality.markdown:
```diff
@@ -58,7 +58,7 @@ quality” – “стоимость качества”). Можно сказа
 программного обеспечения является постоянным объектом заботы программной
 инженерии и обсуждается во многих областях знаний (что вполне обосновано, если
 учесть поистине катастрофический уровень проваленных проектов и
-неудовлетворенность пользователей программных продуктов, ставшая притчей во
```
Please make a single commit with fixed typos.

js-denton replied:
Я не понял в чём ошибка? В слове "неудовлетворённость" была замена "е" на "ё".

ligurio replied:
Куча коммитов с опечатками, сделайте один коммит

js-denton replied:
Есть https://github.com/ligurio/swebok-2004-in-russian/pull/2/commits/e8990df0a0e11d8957b46a8a870acbe2045d4982

ligurio replied:
спасибо

ligurio reviewed (commented):
Thanks for the changes! Please see my comments inline.

ligurio commented:
Вообщем-то бОльшая часть изменений осмысленна, но надо привести коммиты в порядок (как в комментах писал - один коммит - одно изменение: опечатки, сборка fb2, фиксы списков etc) и можно будет влить.

js-denton committed:
Removed a duplicate section

js-denton committed:
Updated pandoc option

* Removed `--self-contained`, key is deprecated, see [1]
* Removed `-s`, needed only for html.
  If build epub this key used by default, see [2]
* Return depth the table of contents by default (3).

1. https://pandoc.org/MANUAL.html#options-affecting-specific-writers
2. https://pandoc.org/MANUAL.html#general-writer-options

js-denton committed:
Updated build epub and html

* Transfered the cover book to title.txt
* Removed style.css file because default style decide better results
* Added keys "--embed-resources --standalone" for build html

js-denton committed:
Added build fb2 book format

ligurio reviewed (commented):
опечатка в коммите "Updated build epub and html": `titel` -> `title`.

ligurio commented on Makefile:
```diff
@@ -20,11 +20,11 @@ NAME = swebok_2004_ru
 
 all: $(NAME).html $(NAME).epub
 
-$(NAME).epub: $(MAKRDOWN_FILES) style.css title.txt
```
Не надо зависимости убирать. Напротив таргета `$(NAME).epub` написаны файлы, от которых таргет зависит. Они никак не дублируют `$(PANDOC_OPT)`.

js-denton replied:
Исправил a8d997ccd3c6b370d10cfd3b3696fd84793278fe

ligurio reviewed (commented):
опечатка в коммите "Updated pandoc option": buld -> build

ligurio reviewed (commented):
коммит Updated pandoc option:

- опечатка contens  -> contents

> https://ieeecs-media.computer.org/media/education/swebok/swebok-v3.pdf

это не оригинальная книга. v3 вышла в 2014, а это перевод SWEBOK от 2004 года

ligurio commented on 0_introduction.markdown:
```diff
@@ -81,11 +81,13 @@ C 1993 года IEEE и ACM координируют свои работы в р
+
+[^4]: Концепция “общепринятости” - generally accepted – определена в IEEE
 Std 1490-1998, Adoption of PMI Standard — A Guide to the Project Management
-Body of Knowledge)
```
опечатки в другом коммите

js-denton replied:
Исправлено a20d24c8022677c328448a86f171a629224ecd86

ligurio commented on 0_introduction.markdown:
```diff
@@ -178,15 +180,15 @@ SWEBOK – “секций” и дает декомпозицию каждой
-соответствующих глав SWEBOK включает расширения (замечания и комментарии),
-помеченные цветом, отличным от цвета перевода оригинального содержания
-SWEBOK.**
+> Представленный перевод SWEBOK 2004 никак не заменяет первоисточника и создан
+> [Сергеем Орликом](https://www.linkedin.com/in/sorlik) при участии [Юрия
+> Булуя](http://www.linkedin.com/in/yurybuluy) без какой-либо поддержки IEEE или
+> других структур по собственной инициативе в полном соответствии со SWEBOK
+> Copyright © 2004 by The Institute of Electrical and Electronics Engineers, Inc.
+> All rights reserved. Copyright and Reprint Permissions: This document may be
+> copied, in whole or in part, in any form or by any means, as is, or with
+> alterations, provided that (1) alterations are clearly marked as alterations
+> and (2) this copyright notice is included unmodified in any copy. Далее перевод
+> соответствующих глав SWEBOK включает расширения (замечания и комментарии),
+> помеченные цветом, отличным от цвета перевода оригинального содержания
+> SWEBOK.
```
коммит про сноски, а здесь форматирование меняется.
насколько помню, угловые скобки применяются для цитирования, а здесь нет цитаты. что именно хотелось сделать в этом изменении?

js-denton replied:
Хотелось выделить как "Заметки" в книгах. Использование "цитирования" к этому лучше всего подходит так как весь фрагмент выделяется.

ligurio replied:
окей, тогда в другом коммите пожалуйста

js-denton replied:
Есть a6f797ca89c971f583c1f42f6fc7862b07b34f53

ligurio commented on 6_software_configuration_management.markdown:
```diff
@@ -390,7 +390,7 @@ SCM-процесс.
 необходимо введение соответствующих полномочий и назначение обязанностей по
 контролю выполнения задач SCM. Аналогичные полномочия и обязанности по надзору
-над SCM-процессом могут существовать в контексте SQA-деятельности. 
+над SCM-процессом могут существовать в контексте SQA-деятельности.
```
это не про сноски изменение

js-denton replied:
Исправлено 7f030c4003532333429dde4192eee041bb935079

ligurio reviewed (commented):
коммит " Removed a dublicate section ": опечатка dublicate -> duplicate

ligurio commented on 9_software_engineering_tools_and_methods.markdown:
```diff
@@ -399,7 +399,7 @@ methods). Такие специализированные методы разр
 Языки спецификаций могут быть ориентированы на модель, свойства и поведение. По
 мнению автора, ярким примером такого рода методов являются формальные методы
-описания требований, интерес к которым периодически возникает на протяжении
+описания требований, интерес к которым периодически возникает на протяжениизз
```
внесена опечатка: протяжениизз

js-denton replied:
Откатано a20d24c8022677c328448a86f171a629224ecd86

ligurio commented on Makefile:
```diff
@@ -18,11 +18,14 @@ PANDOC_OPT += title.txt $(MAKRDOWN_FILES)
+	$(PANDOC) $(PANDOC_OPT) -o $(NAME).fb2
+
 $(NAME).html:
 	$(PANDOC) $(PANDOC_OPT) --embed-resources --standalone -o $(NAME).html
```
в таргеты ` release` и `clean` надо тоже добавить

js-denton replied:
Доработано dfaa60a4b96798a67a70502370ec7207706553c4

RastaDill commented:
> коммит Updated pandoc option:
> 
> * опечатка contens  -> contents
> 
> > https://ieeecs-media.computer.org/media/education/swebok/swebok-v3.pdf
> 
> это не оригинальная книга. v3 вышла в 2014, а это перевод SWEBOK от 2004 года



> коммит Updated pandoc option:
> 
> * опечатка contens  -> contents
> 
> > https://ieeecs-media.computer.org/media/education/swebok/swebok-v3.pdf
> 
> это не оригинальная книга. v3 вышла в 2014, а это перевод SWEBOK от 2004 года

Да, точно. Я [нашёл оригинал](https://www.inf.ed.ac.uk/teaching/courses/seoc1/2005_2006/resources/SWEBOK_2004.pdf), так там вообще уровень вложенности 1 :-)
Я наверное лучше удалю часть комментария с книгой.

ligurio commented:
> Да, точно. Я [нашёл оригинал](https://www.inf.ed.ac.uk/teaching/courses/seoc1/2005_2006/resources/SWEBOK_2004.pdf), так там вообще уровень вложенности 1 :-)

У меня немного другой [^1], но они должны быть одинаковы по содержанию.

[^1]: https://bronevichok.ru/static/SWEBOK.pdf

js-denton committed:
Highlighted the notes about the translation (almost like in the book)

ligurio commented:
Поправил еще одну опечатку:

```diff
--- a/software_lifecycle_models.markdown
+++ b/software_lifecycle_models.markdown
@@ -163,7 +163,7 @@ Processes” – “Процессы жизненного цикла прогр
 
 ### Основные процессы жизненного цикла - Primary Processes
 
-5.1 Заказ - Acqusition
+5.1 Заказ - Acquisition
 5.2 Поставка - Supply
 5.3 Разработка - Development
 5.4 Эксплуатация - Operation
```

и поменял формат ссылок в коммит мессадже "Updated pandoc option".

и поправил это: 
> коммит " Removed a dublicate section ": опечатка dublicate -> duplicate

ligurio commented:
@RastaDill спасибо!
